### DISCUSSION TITLE
How to determine Elo of chess AI?

### DISCUSSION THREAD

**Discussion by jamesbraza** [official-stockfish/Stockfish#4446] [Q&A] [answered]
How can one determine an AI-based chess player's Elo?

Here is a basic idea from https://github.com/niklasf/python-chess/issues/978, but I figure someone here might know a better way:

> 1. Instantiate a `StockfishPlayer` of known Elo
> 2. Play against it, and update Elo accordingly
> ...
> 3. If Elo moves a significant amount across matches, return to step 1 with an updated Elo
> 4. Otherwise return the `StockfishPlayer`'s Elo, possibly with confidence intervals

**Comment by dubslow** [comment]
playing games is the short answer. playing *lots* of games -- tens or hundreds of thousands -- is a slightly better answer.

see for example https://github.com/official-stockfish/Stockfish/wiki/Regression-Tests

**Comment by DrGrekenstein** [comment, marked as answer]
https://en.wikipedia.org/wiki/Elo_rating_system

Here is the basic formula:

![image](https://github.com/official-stockfish/Stockfish/assets/158576766/706ccbeb-5357-45cb-97a9-3211f0161d46)


![image](https://github.com/official-stockfish/Stockfish/assets/158576766/d4626dc2-2912-48f5-9451-912cb6a5db5a)
